app-text/wordgrinder/wordgrinder-0.7.1.recipe

```diff
@@ -11,40 +11,57 @@ REVISION="1"
 SOURCE_URI="https://github.com/davidgiven/wordgrinder/archive/$portVersion.tar.gz"
 CHECKSUM_SHA256="54085af7963e1f67342bc0b1b20d1ccc75494f2e23d401b601cf3089acea747c"
 
-ARCHITECTURES="!x86_gcc2 !x86_64"
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
 
 PROVIDES="
-	wordgrinder = $portVersion
-	cmd:WordGrinder = $portVersion
+	wordgrinder$secondaryArchSuffix = $portVersion
+	cmd:wordgrinder$commandSuffix = $portVersion
 	"
 REQUIRES="
-	haiku
-	lib:libncurses
-	lib:liblua
-	lib:libz
+	haiku$secondaryArchSuffix
+#	lib:libfreetype$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.3
+	lib:libncurses$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
 	"
 
 BUILD_REQUIRES="
-	haiku_devel
-	devel:libncurses
-	devel:liblua
-	devel:libz
+	haiku${secondaryArchSuffix}_devel
+#	devel:libfreetype$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.3
+	devel:libncurses$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
-	cmd:g++
+	cmd:cc$secondaryArchSuffix
 	cmd:make
 	cmd:ninja
-	cmd:pkg_config
+	cmd:pkg_config$secondaryArchSuffix
 	"
 
+defineDebugInfoPackage wordgrinder$secondaryArchSuffix \
+	$commandBinDir/wordgrinder
+
 BUILD()
 {
-	make OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
-		DOCDIR=$docDir MANDIR=$manDir
+	make PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
+		MANDIR=$manDir HOME="`finddir B_USER_DIRECTORY`" \
+		LUA_PACKAGE=lua
 }
 
 INSTALL()
 {
-	make install OBJDIR=$sourceDir/build PREFIX=$prefix BINDIR=$binDir \
-		DOCDIR=$docDir MANDIR=$manDir
+	make install PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
+		MANDIR=$manDir HOME="`finddir B_USER_DIRECTORY`" \
+		LUA_PACKAGE=lua
+	rm -r $docDir/wordgrinder
```

Comment:
Oops, you dropped the README.wg file.
```diff
+	mv $docDir/wordgrinder/README.wg $docDir
-	rm -r $docDir/wordgrinder
+ 	rmdir $docDir/wordgrinder
 }
```

BTW, regarding your previous _"variant"_ of HOME:
```
		MANDIR=$manDir HOME=$`finddir B_USER_DIRECTORY` \
```
> Ah, k, will change, didn't produce an error here :)

The build did not fail because HOME was being set to something (`HOME=$/boot/home`) but that would not have been very useful ;-)

Reply:
WordGrinder doesn't have any knowledge of README.wg, so feel free to move it. so I thought it wasn't needed :)

Reply:
OK, I see.
Well, what the author was saying is that the program does not expect that file to be anywhere.
But it is still nice to have it in $docDir.
Great, I saw you just added it back :-)

Reply:
Trying to please the master ;)

Reply:
ps, there are other examples in the source, maybe they could be moved to $docDir too?